bigbluebutton-html5/imports/ui/components/plugins-engine/types.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import * as PluginSdk from 'bigbluebutton-html-plugin-sdk';
+
+export interface PluginProvidedStateProps {
+    uuid: string
+}
+
+export interface PluginsEngineComponentProps {
+    containerRef: React.RefObject<HTMLDivElement>
+}
+
+export interface PluginLoaderComponentProps {
+    uuid: string
+    containerRef: React.RefObject<HTMLDivElement>
+    loadedPlugins: React.MutableRefObject<number>
+    setLastLoadedPlugin: React.Dispatch<React.SetStateAction<HTMLScriptElement | undefined>>
+    pluginConfigSettings: PluginConfigSetting
+}
+
+export interface PluginEngineProps {
+    onReady: Function
+}
+
+export interface PluginConfigSetting {
+    name: string
+    url: string
+}
+
+export interface UniquePluginConfigSetting {
+    name: string
+    url: string
+    uuid: string
+}
+
+export interface PluginObjects {
+    whiteboardToolbarItems: PluginSdk.WhiteboardToolbarItem[]
+}
+
+export type PluginProvidedStateStaticData = {
+    [uuid: string]: PluginObjects
+}
```

Comment:
This is the type represents the map containing the state provided by each plugin, we keep track of this information so we can compute the aggregate state (see @PluginsProvidedState).

Said that, please rename it to: `PluginsProvidedStateMap`

And add a tsdoc specifying what this type is meant for (just like I did on this comment).